lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Server.hs

```diff
@@ -68,25 +68,30 @@ api :: Proxy API
 api = Proxy
 
 server
-    :: Tracer IO String
+    :: Tracer IO ()
```

Comment:
Was curious: did you consider using `TVar IO ()` or I guess `TVar IO WalletState`?  

And taking the latter one step further, we get the question whether we could have `DBVar STM WalletState` instead of in `IO` 🤔 cc @HeinrichApfelmus

Reply:
For the way it is used ATM, the STM composability is not a requirement. We are serving this signal as a long poll over the network. It's also dangerous for the signal producer which could get stuck (in theory) by other concurrent producers. `STM ()` could be better ? But I would not dig into that until we have a real need.